### PR TITLE
Lesson 17

### PR DESCRIPTION
**Pull request by vtx123**

### CODE REVIEW

File: Lesson17/Vector2d.cpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+float& Vector2d::operator[](int i) {
+    return i > 0 ? _y : _x;
+}
+
+Vector2d& Vector2d::operator=(const Vector2d & vec2d) {
+    _x = vec2d._x;
+    _y = vec2d._y;
+
+    return *this;
+}
```
Comment (helen-black) [inline Lesson17/Vector2d.cpp]: check if they aren't equal

File: Lesson17/Vector2d.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+float Vector2d::operator()(){
+    return static_cast<float>(sqrt(pow(_x, 2) + pow(_y, 2)));
+}
+
+float& Vector2d::operator[](int i) {
+    return i > 0 ? _y : _x;
+}
+
```
Comment (helen-black) [inline Lesson17/Vector2d.cpp]: check if i is nou out of range

**Review by helen-black — CHANGES_REQUESTED** [review]

File: Lesson17/Vector2d.cpp (hunk 2 above)
Comment (vtx123) [reply Lesson17/Vector2d.cpp]: smth like this?
float& Vector2d::operator[](int i) {
    if(i <= 0 ) { return _x;} else { return _y}
}

File: Lesson17/Vector2d.cpp (hunk 1 above)
Comment (vtx123) [reply Lesson17/Vector2d.cpp]: fixed

**Pull request merged**